lib/rubocop.rb

```diff
@@ -547,6 +547,13 @@
 require_relative 'rubocop/cop/rails/has_and_belongs_to_many'
 require_relative 'rubocop/cop/rails/has_many_or_has_one_dependent'
 require_relative 'rubocop/cop/rails/http_positional_arguments'
+
+begin
+  require_relative 'rubocop/cop/rails/http_status'
+rescue LoadError # rubocop:disable Lint/HandleExceptions
+  # Rails/HttpStatus cannot be loaded if rack/utils is unavailable.
+end
```

Comment:
I think the rescuing should be in `lib/rubocop/cop/rails/http_status.rb`.
And RuboCop should disable the cop or disable auto-correct if RuboCop cannot load `rack/utils`. 
For example:

```
begin
  require 'rack/utils'
rescue LoadError
end
```


Because `RuboCop::Cop::Rails::HttpStatus` is not defined when RuboCop cannot load `rack/utils` currently. I think it is confusing.

Reply:
Agree. No reason to hoist this rescue up here. 🙂

Reply:
Yes but this cop does not have reason to live outside a Rails application no ?
I understand if rack cannot be found the autocorrect won’t work but what about the
offense error message ? Should we display a default message such as [this](https://github.com/backus/rubocop-rspec/blob/5787a9056db43e196c2f597e4fa289082cf5b555/lib/rubocop/cop/rspec/rails/http_status.rb#L79-L80) and [this](https://github.com/backus/rubocop-rspec/blob/5787a9056db43e196c2f597e4fa289082cf5b555/lib/rubocop/cop/rspec/rails/http_status.rb#L118-L119) ?

Is this approach ([here](https://github.com/backus/rubocop-rspec/blob/5787a9056db43e196c2f597e4fa289082cf5b555/lib/rubocop/cop/rspec/rails/http_status.rb#L32-L37) and [here](https://github.com/backus/rubocop-rspec/blob/5787a9056db43e196c2f597e4fa289082cf5b555/lib/rubocop/cop/rspec/rails/http_status.rb#L53-L55)) what you have in mind ? 

Thanks !

Reply:
I changed the similar cop in rubocop-rspec to have a hard dependency on Rack, which simplified things a great deal: https://github.com/backus/rubocop-rspec/commit/3e4d6e6385fea65a6becfffc51361ec1fa89468e#diff-1c69b4694ea24f115dacb59c8911ee45

I think it is reasonable to assume that when rubocop is being used with Rails cops enabled, you have `rack` available in the load path. And if not, we just don’t define/load this cop. (Or will that conflict with it being named in config/default.yml?)

Reply:
Yeah, that's true. At any rate - I don't want us to have conditional logic in this particular file.